src/main/resources/io/federecio/dropwizard/swagger/index.ftl

```diff
@@ -62,19 +62,44 @@
             }
         }
 
+      function addApiKeyKrakenAuthorization() {
+        var key = $('#input_apiKeyKraken')[0].value;
+        if(key && key.trim() != "") {
+            swaggerUi.api.clientAuthorizations.add("api_key", new SwaggerClient.ApiKeyAuthorization("x-auth-id", key, "header"));
+        }
+      }
+
+      function addApiSecretKrakenAuthorization() {
+        var secret = $('#input_apiSecretKraken')[0].value;
+        if(secret && secret.trim() != "") {
+            swaggerUi.api.clientAuthorizations.add("api_secret", new SwaggerClient.ApiKeyAuthorization("x-auth-key", secret, "header"));
+        }
+      }
+
+
+
       $('#input_apiKey').change(function() {
         addApiKeyAuthorization();
       });
 
+      $('#input_apiSecretKraken').change(function() {
+        addApiSecretKrakenAuthorization();
+      });
+
+      $('#input_apiKeyKraken').change(function() {
+        addApiKeyKrakenAuthorization();
+      });
+
       $('#input_authHeader').change(function() {
           addAuthorizationHeader();
       });
 
       $('#input_headerSelect').change(function() {
+          $('#header_0').hide();
+          $('#header_1').hide();
+          $('#header_2').hide();
           var toShow = $( this ).val();
           $('#header_'+toShow).show();
```

Comment:
I didn't modify this line. Please read the description 
> I didn't refactor code on purpose so that there are mostly additions so we can later rebase the forked repo from origin.